collector/topological_inventory.py

```diff
@@ -214,25 +219,40 @@ def worker(_: str, source_id: str, dest: str, acct_info: dict) -> None:
     account_id = acct_info['account_id']
     headers = {"x-rh-identity": b64_identity}
 
-    if ALL_TENANTS:
-        resp = utils.retryable('get', TENANTS_URL, headers=headers)
-        tenants_headers = \
-            [tenant_header_info(t["external_tenant"]) for t in resp.json()]
-
-        LOGGER.info('Fetching data for ALL(%s) Tenants', len(tenants_headers))
-
-        for tenant_header in tenants_headers:
-            LOGGER.debug('%s: ---START Account# %s---',
-                         thread.name, tenant_header['acct_no'])
-            headers = tenant_header['headers']
-            topological_inventory_data(_, source_id, dest, headers, thread)
-            utils.set_processed(tenant_header['acct_no'])
-            LOGGER.debug('%s: ---END Account# %s---',
-                         thread.name, tenant_header['acct_no'])
-    else:
-        LOGGER.info('Fetching data for current Tenant')
-        topological_inventory_data(_, source_id, dest, headers, thread)
-        utils.set_processed(account_id)
+    with DATA_COLLECTION_TIME.time():
```

Comment:
How about wrap the `worker()` in this `time()` decorator?

Reply:
That is how I had it first, but something was off with the  `aiops_data_collector_data_collection_time_sum` value

To confirm the above, I tested it again with `time()` decorator around `worker`, and these are the values that I'm seeing, which are clearly incorrect -

```
aiops_data_collector_data_collection_time_count 3.0
aiops_data_collector_data_collection_time_sum 1.292303204536438e-05
```

So, not going to change the current `time` decorator setup.

Reply:
Resolved this with `@DATA_COLLECTION_TIME.time()` wrapper around `worker`

Thanks for your help @jameswnl !